From f4f3d93fe8c95881907ebbd61407e0f5d16ef179 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Sat, 25 Jul 2026 09:51:38 +0200
Subject: [PATCH 1/4] Skip linking cfitsio and healpix_cxx programs for Pyodide
 builds

They fail to link with duplicate symbols and are unneeded.
---
 setup.py | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index e727611..26b0d76 100755
--- a/setup.py
+++ b/setup.py
@@ -7,7 +7,7 @@ import sys
 import shlex
 import shutil
 from Cython.Distutils import build_ext
-from sysconfig import get_config_vars
+from sysconfig import get_config_vars, get_platform
 from subprocess import check_output, CalledProcessError, check_call
 from setuptools import setup, Extension
 from setuptools.command.build_clib import build_clib
@@ -27,6 +27,10 @@ if "test" in sys.argv:
     sys.exit(1)
 
 
+def is_emscripten():
+    return get_platform().startswith("emscripten")
+
+
 class build_external_clib(build_clib):
     """Subclass of Distutils' standard build_clib subcommand. Adds support for
     libraries that are installed externally and detected with pkg-config, with
@@ -179,6 +183,13 @@ class build_external_clib(build_clib):
 
             # Run make install.
             cmd = ["make", "install"]
+
+            # On Emscripten, these binaries fail to link with duplicate symbols.
+            # healpy only needs the libraries and headers, so skip the programs
+            # entirely by emptying the automake variables that list them.
+            if is_emscripten():
+                cmd += ["bin_PROGRAMS=", "noinst_PROGRAMS="]
+
             log.info("%s", " ".join(cmd))
             check_call(cmd, cwd=build_temp, env=env)
 
-- 
2.54.0

